### PR TITLE
Feat: add options parameter to addMessageToChatHistory

### PR DESCRIPTION
**Pull request by changangus**
## Description

Adds options parameter to addMessageToChatHistory function.

## Spec

See Issue: [FSA21V2-60](https://sparkbox.atlassian.net/browse/FSA21V2-60)

## Validation

- [x] This PR has code changes, and our linters still pass.
- [x] This PR has new code, so new tests were added or updated, and they pass.

### CODE REVIEW

File: src/js/chatWindow.js [hunk 1]
```diff
@@ -1,8 +1,20 @@
-export const addMessageToChatHistory = (m) => {
+const guardOptions = (options) => {
+  if (options.messageType !== 'outgoing' && options.messageType !== 'incoming') {
+    return false;
+  }
+
+  return true;
+};
```
Comment (colbyallenc) [inline src/js/chatWindow.js:9]: I love how you included this guard to throw an error messages that aren't explicitly 'incoming' or 'outgoing'. Very nice!

File: src/js/chatWindow.js [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
+
+export const addMessageToChatHistory = (m, options = { messageType: 'outgoing' }) => {
   if (typeof m !== 'string' || m === '') return;
 
+  if (!guardOptions(options)) throw new Error('Message Type is incorrect');
```
Comment (colbyallenc) [inline src/js/chatWindow.js:19]: 🏆

File: src/js/chatWindow.js [hunk 3]
```diff
@@ -1,8 +1,20 @@
...
+
   const c = document.createElement('div');
-  c.classList.add('chat-window__message', 'chat-window__message--outgoing');
+
+  c.classList.add('chat-window__message', `chat-window__message--${options.messageType}`);
```
Comment (colbyallenc) [inline src/js/chatWindow.js]: is there a more descriptive variable that we can use for `c`?

File: src/js/chatWindow.js [hunk 4]
```diff
@@ -1,8 +1,20 @@
-export const addMessageToChatHistory = (m) => {
+const guardOptions = (options) => {
```
Comment (colbyallenc) [inline src/js/chatWindow.js:1]: Do you plan to write tests around this guard? I'm not sure how valuable they'll be but just curious.

File: src/js/chatWindow.js [hunk 5]
```diff
@@ -1,8 +1,20 @@
...
+  if (options.messageType !== 'outgoing' && options.messageType !== 'incoming') {
+    return false;
+  }
+
+  return true;
+};
+
+export const addMessageToChatHistory = (m, options = { messageType: 'outgoing' }) => {
```
Comment (colbyallenc) [inline src/js/chatWindow.js]: is there a more descriptive variable name that we can use for `m`?

File: src/js/chatWindow.spec.js [hunk 6]
```diff
@@ -37,3 +37,30 @@ test('sanitizes input and adds valid messages to the chat window', () => {
   expect(document.getElementsByClassName('chat-window__messages')[0].lastChild)
     .toStrictEqual(expectedNode);
 });
+
+describe('testing options parameter functionality in addMessageToChatHistory', () => {
+  it('throws an error if messageType is not "incoming" or "outgoing"', () => {
+    expect(() => { addMessageToChatHistory('testing', { messageType: 'unknown' }); }).toThrowError('Message Type is incorrect');
+  });
```
Comment (colbyallenc) [inline src/js/chatWindow.spec.js:52]: I'd consider this to be a nice integration test. It tests the functionality of the guard within the `addMessageToChatHistory` function. Nice!

**Review by colbyallenc — COMMENTED** [review]
- 

Great job @changangus! I really loved that you were able to create that guard after we last spoke! I also loved the way that you pulled that logic out and called it within that function.

File: src/js/chatWindow.js (hunk 3 above)
Comment (changangus) [reply src/js/chatWindow.js]: Yep! I changed it to chatMessageElement

File: src/js/chatWindow.js (hunk 5 above)
Comment (changangus) [reply src/js/chatWindow.js]: and I changed this one to "message"!

**Review by colbyallenc — APPROVED** [review]

**Pull request merged**